middle_end/flambda2.0/simplify/simplify_variadic_primitive.ml

```diff
@@ -29,51 +28,47 @@ let simplify_make_block dacc _prim dbg
     let env_extension = TEE.one_equation (Name.var result_var) ty in
     Reachable.invalid (), env_extension, dacc
   in
-  match make_block_kind with
-  | Full_of_values (tag, value_kinds) ->
-    if List.compare_lengths value_kinds args <> 0 then begin
-      (* CR mshinwell: improve message *)
-      Misc.fatal_errorf "GC value_kind indications in [Make_block] don't \
-          match up 1:1 with arguments: %a"
-        Simple.List.print args
-    end;
-    (* CR mshinwell: This could probably be done more neatly. *)
-    let found_bottom = ref false in
-    let fields =
-      assert (List.compare_lengths value_kinds args_with_tys = 0);
-      List.map2 (fun ((arg : Simple.t), arg_ty) _value_kind ->
-          if T.is_bottom (DE.typing_env denv) arg_ty then begin
-           found_bottom := true
-          end;
-          Simple.pattern_match arg
-            ~const:(fun _ -> arg_ty)
-            ~name:(fun name -> T.alias_type_of K.value (Simple.name name)))
-        args_with_tys value_kinds
+  if List.compare_lengths shape args <> 0 then begin
+    (* CR mshinwell: improve message *)
+    Misc.fatal_errorf "GC value_kind indications in [Make_block] don't \
+        match up 1:1 with arguments: %a"
+      Simple.List.print args
+  end;
+  (* CR mshinwell: This could probably be done more neatly. *)
+  let found_bottom = ref false in
+  let fields =
+    assert (List.compare_lengths shape args_with_tys = 0);
```

Comment:
You could remove that assert, it is redundant with the check a few lines above

Reply:
Done